lib/swoosh/adapters/mailgun.ex

```diff
@@ -27,10 +27,20 @@ defmodule Swoosh.Adapters.Mailgun do
 
   def deliver(%Email{} = email, config \\ []) do
     headers = prepare_headers(email, config)
-    params = email |> prepare_body |> Plug.Conn.Query.encode
     url = [base_url(config), "/", config[:domain], @api_endpoint]
 
-    case :hackney.post(url, headers, params, [:with_body]) do
+    payload = case prepare_body(email) do
```

Comment:
What do you think of moving this case statement into a function at the end of `prepare_body/1`?

Not too bothered but maybe it would be cleaner?